cli/src/command/wallet.rs

```diff
@@ -128,32 +128,41 @@ pub async fn init_command(
     snapshot_path: &Path,
     parameters: InitParameters,
 ) -> Result<Wallet, Error> {
-    let password = get_password("Stronghold password", !snapshot_path.exists())?;
-    let secret_manager = SecretManager::Stronghold(
-        StrongholdSecretManager::builder()
-            .password(&password)
-            .build(snapshot_path)?,
-    );
-    let wallet = Wallet::builder()
-        .with_secret_manager(secret_manager)
-        .with_client_options(ClientOptions::new().with_node(parameters.node_url.as_str())?)
-        .with_storage_path(storage_path.to_str().expect("invalid unicode"))
-        .with_coin_type(parameters.coin_type)
-        .finish()
-        .await?;
+    if storage_path.exists() {
+        return Err(Error::Miscellaneous(format!(
+            "cannot initialize: {} already exists",
+            storage_path.display()
+        )));
+    }
+    if snapshot_path.exists() {
+        return Err(Error::Miscellaneous(format!(
+            "cannot initialize: {} already exists",
+            snapshot_path.display()
+        )));
+    }
+    // Set up password.
+    let password = get_password("Stronghold password", true)?;
 
+    // Set up mnemonic.
     let mnemonic = match parameters.mnemonic_file_path {
         Some(path) => import_mnemonic(&path).await?,
         None => enter_or_generate_mnemonic().await?,
     };
 
-    if let SecretManager::Stronghold(secret_manager) = &mut *wallet.get_secret_manager().write().await {
-        secret_manager.store_mnemonic(mnemonic).await?;
-    } else {
-        panic!("cli-wallet only supports Stronghold-backed secret managers at the moment.");
-    }
+    let secret_manager = StrongholdSecretManager::builder()
+        .password(&password)
+        .build(snapshot_path)?;
+    // TODO: allow providing the mnemonic in the builder?
```

Comment:
We have an issue to remove TODOs so let's not add some more. Open an issue if you want to discuss that.

Reply:
Done